sdcm/utils/gce_builder.py

```diff
@@ -0,0 +1,239 @@
+
+from sdcm.utils.gce_region import GceRegion
+from sdcm.utils.gce_utils import SUPPORTED_PROJECTS, SUPPORTED_REGIONS
+from sdcm.sct_runner import GceSctRunner
+from sdcm.keystore import KeyStore
+
+LOGGER = logging.getLogger(__name__)
+
+JENKINS_CONFIG_FORMAT = """
+// just modify this config other code just logic
+config = [
+    name: "{name}",
+    region: "{region_name}",
+    projectId: "{project_id}",
+    credentialsId: "{project_id}",
+    instanceCapStr: "{max_instances}",
+    labels: "{jenkins_labels}",
+]
+
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud
+import com.google.jenkins.plugins.computeengine.InstanceConfiguration
+import com.google.jenkins.plugins.computeengine.SshConfiguration
+import hudson.plugins.sshslaves.SSHConnector
+import hudson.plugins.sshslaves.verifiers.NonVerifyingKeyVerificationStrategy
+import hudson.model.*
+import jenkins.model.Jenkins
+
+ComputeEngineCloud gcpCloud = new ComputeEngineCloud(config.name, config.projectId, config.credentialsId, config.instanceCapStr)
+gcpCloud.setNoDelayProvisioning(true)
+
+InstanceConfiguration instanceConfiguration = new InstanceConfiguration()
+instanceConfiguration.setNumExecutorsStr("4")
+instanceConfiguration.setOneShot(false)
+instanceConfiguration.setNamePrefix("builders")
+instanceConfiguration.setDescription("SCT Jenkins builders")
+instanceConfiguration.setRegion("https://www.googleapis.com/compute/v1/projects/{project_id}/regions/{region_name}")
+instanceConfiguration.setZone("https://www.googleapis.com/compute/v1/projects/{project_id}/zones/{region_name}-b")
+instanceConfiguration.setLabelString(config.labels)
+instanceConfiguration.setMode(Node.Mode.NORMAL)
+instanceConfiguration.setTemplate("https://www.googleapis.com/compute/v1/projects/{project_id}/global/instanceTemplates/{name}")
+instanceConfiguration.setRetentionTimeMinutesStr("6")
+instanceConfiguration.setLaunchTimeoutSecondsStr("300")
+instanceConfiguration.setRunAsUser("jenkins")
+instanceConfiguration.setSshConfiguration(SshConfiguration.builder().customPrivateKeyCredentialsId("6e17c350-0a47-4a59-b0b9-7e64f1c35a65").build())
+
+gcpCloud.addConfiguration(instanceConfiguration)
+
+// get Jenkins instance
+Jenkins jenkins = Jenkins.get()
+
+// clear old configuration
+jenkins.clouds.removeAll {{ it.name == "gce-${{config.name}}" }}
+
+res = jenkins.clouds.add(gcpCloud)
+
+// save current Jenkins state to disk
+jenkins.save()
+"""
+
+
+class GceBuilder:
+    """
+    This class is for configuration our Jenkins setup for GCE
+
+    It created a launch template based on sct-runner image, and adds configuration needed in Jenkins to use it
+    """
+
+    def __init__(self, region: GceRegion, number=1):
+        self.region = region
+        self.number = number
+        self.jenkins_info = KeyStore().get_json("jenkins.json")
+        self.runner = GceSctRunner(region_name=self.region.region_name,
+                                   availability_zone="a")
```

Comment:
Why we want to stick to a specific az? Is it a problem to allow to set it?

Reply:
it's pointless, we pass it along only since `GceSctRunner` mandate it.
we using this class only to extract the image name of the sct-runner to put in the launch template